src/odemis/gui/cont/fastem_acq.py

```diff
@@ -581,32 +563,231 @@ def _on_alignment_done(self, future):
+        """
+        self.button.Enable(not mode)
+
+
+class FastEMScintillatorCalibrationController(FastEMCalibrationController):
+    """
+    Controls the selection panel for the individual scintillators, the calibration button to start the
+    calibration and the process in the calibration panel in the FastEM acquisition tab.
+    """
+    def __init__(self, tab_data, tab_panel, calib_prefix, calibrations):
+        """
+        :param tab_data: (FastEMAcquisitionGUIData) The representation of the microscope GUI.
+        :param tab_panel: (wx.Frame) The calibration panel, which contains the selection buttons
+                          for the individual scintillators, the calibration button to start the
+                          calibration, and the gauge and the label of the gauge to indicate the progress.
+        :param calib_prefix: (str) A prefix, which can indicate the order/type of the calibration (e.g. "calib_1").
+        :param calibrations: (list[Calibrations]) List of calibrations that should be run.
+        """
+        super().__init__(tab_data, tab_panel, calib_prefix, calibrations)
+
+        # listen to calibration regions selected/deselected and update estimated calibration time accordingly
+        for roc in tab_data.calibration_regions.value.values():
+            roc.coordinates.subscribe(self._on_calibration_state)
+
+        self._main_data_model.active_scintillators.subscribe(self._on_calibration_state)
+
+    def calibrate(self, evt):
+        """
+        Start or cancel the calibrations for all set regions of calibrations (roc) when the calibration
+        button is triggered. A progressive future for all rocs is created.
+        :param evt: (GenButtonEvent) Button triggered.
+        """
+        # check if cancelled
+        if self._tab_data.main.is_acquiring.value:
+            logging.debug("Calibration was cancelled.")
+            align_fastem._executor.cancel()  # all the rest will be handled by on_alignment_done()
+            # FIXME cancelling does not yet display the correct label in the gui
+            return
+
+        # calibrate
+        self._tab_data.main.is_acquiring.unsubscribe(self._on_is_acquiring)
+        # Briefly unsubscribe as don't need to know about this event (is_acquiring = True):
+        # It  would disable the calibration button, but want to be able to still cancel calibration.
+        self._tab_data.main.is_acquiring.value = True  # make sure the acquire/tab buttons are disabled
+        self._tab_data.main.is_acquiring.subscribe(self._on_is_acquiring)
+
+        self._on_calibration_state()  # update the controls in the panel
+
+        futures = {}
+        for roc_num in sorted(self._tab_data.calibration_regions.value.keys()):
+            # check if calibration region (ROC) on scintillator is set
+            roc = self._tab_data.calibration_regions.value[roc_num]
+            if roc.coordinates.value != stream.UNDEFINED_ROI:
```

Comment:
The comment on line 668 is not completely clear to me with respect line. Can you clarify that comment or add another comment here, explaining what `roc.coordinates.value != stream.UNDEFINED_ROI` means. Does it mean the ROC is not set?

Reply:
`roc.coordinates.value != stream.UNDEFINED_ROI` means that the ROC is set, if it is undefined, it is not set.